wallet/src/main/scala/org/bitcoins/wallet/DLCWallet.scala

```diff
@@ -75,15 +83,34 @@ abstract class DLCWallet extends Wallet {
     }
   }
 
+  private def calcDLCPubKeys(
+      xpub: ExtPublicKey,
+      keyIndex: Int): DLCPublicKeys = {
+    val fundingKey =
+      xpub
+        .deriveChildPubKey(BIP32Path.fromString(s"m/0/$keyIndex"))
+        .get
+        .key
+
+    val payoutKey =
+      xpub
+        .deriveChildPubKey(BIP32Path.fromString(s"m/0/${keyIndex + 1}"))
+        .get
+        .key
+
+    DLCPublicKeys.fromPubKeys(fundingKey,
+                              payoutKey,
+                              networkParameters.asInstanceOf[BitcoinNetwork])
```

Comment:
nit: To avoid any possible issues in the future, you should be able to do
```
networkParameters match {
  case network: BitcoinNetwork => ...
}
```
for exhaustiveness checks